apps/heft/src/plugins/TypeScriptPlugin/Tslint.ts

```diff
@@ -36,28 +36,63 @@ export class Tslint extends LinterBase<TTslint.RuleFailure> {
     this._fileSystem = options.fileSystem;
   }
 
+  private static _resolvePackageMainFilePath(packagePath: string): string {
+    const packageJson: IPackageJson | undefined = this._packageJsonLookup.tryLoadPackageJsonFor(packagePath);
+    if (!packageJson) {
+      throw new Error(
+        `Supplied package path ${packagePath} could not resolve a package.json while searching for main file`
+      );
+    } else if (!packageJson.main) {
+      throw new Error(`Package at path ${packagePath} specifies no main file`);
+    }
+    return path.resolve(packagePath, packageJson.main);
+  }
+
+  /**
+   * getConfigHash returns the sha1 hash unique to the contents of the given config as
+   * well as all the configs that the given one extends.
+   *
+   * If suppied the previousHash parameter, the hash is updated with the contents of the
+   * file's extended configs and itself before being returned. Passing a digested hash to
+   * this parameter will result in an error.
+   */
   public static getConfigHash(
     configFilePath: string,
     terminal: Terminal,
-    fileSystem: IExtendedFileSystem
+    fileSystem: IExtendedFileSystem,
+    previousHash?: crypto.Hash
   ): crypto.Hash {
     interface IMinimalConfig {
-      extends?: string;
+      extends?: string | string[];
     }
 
     terminal.writeVerboseLine(`Examining config file "${configFilePath}"`);
-
+    // if configFilePath is not a json file, assume that it is a package whose main file
+    // is a config file, per the the "extends" spec of tslint.json, found at
+    //  https://palantir.github.io/tslint/usage/configuration/
+    if (!configFilePath.endsWith('.json')) {
+      configFilePath = this._resolvePackageMainFilePath(configFilePath);
+    }
     const rawConfig: string = fileSystem.readFile(configFilePath);
     const parsedConfig: IMinimalConfig = JsonFile.parseString(rawConfig);
-    let hash: crypto.Hash;
-    if (parsedConfig.extends) {
+    const extendsProperty: string | string[] | undefined = parsedConfig.extends;
+    let hash: crypto.Hash = previousHash || crypto.createHash('sha1');
+
+    if (extendsProperty instanceof Array) {
+      for (const extendFile of extendsProperty) {
+        const extendFilePath: string = Import.resolveModule({
+          modulePath: extendFile,
+          baseFolderPath: path.dirname(configFilePath)
+        });
+        hash = this.getConfigHash(extendFilePath, terminal, fileSystem, hash);
+      }
+    } else if (extendsProperty) {
+      // note that if we get here, extendsProperty is a string
       const extendsFullPath: string = Import.resolveModule({
-        modulePath: parsedConfig.extends,
+        modulePath: extendsProperty,
         baseFolderPath: path.dirname(configFilePath)
       });
-      hash = Tslint.getConfigHash(extendsFullPath, terminal, fileSystem);
-    } else {
```

Comment:
Why deleting this condition? What if there is no extend property in the config file?

Reply:
this condition is handled by the update() call in the return line